src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaContractTests.cs

```diff
@@ -59,25 +70,50 @@ public static void ArgumentValidation(SlhDsaAlgorithm algorithm, bool shouldDisp
         }
 
-        public static IEnumerable<object[]> ApiWithDestinationSpanTestData =>
-            from algorithm in SlhDsaTestData.AlgorithmsRaw
-            from destinationLargerThanRequired in new[] { true, false }
-            select new object[] { algorithm, destinationLargerThanRequired };
+        [Theory]
+        [MemberData(nameof(ArgumentValidationData))]
+        public static void ArgumentValidation_PbeParameters(SlhDsaAlgorithm algorithm, bool shouldDispose)
+        {
+            using SlhDsa slhDsa = SlhDsaMockImplementation.Create(algorithm);
+
+            if (shouldDispose)
+            {
+                // Test that argument validation exceptions take precedence over ObjectDisposedException
+                slhDsa.Dispose();
+            }
+
+            SlhDsaTestHelpers.AssertEncryptedExportPkcs8PrivateKey(export =>
+            {
+                // Unknown algorithm
+                AssertExtensions.Throws<CryptographicException>(() => 
+                    export(slhDsa, "password", new PbeParameters(PbeEncryptionAlgorithm.Unknown, HashAlgorithmName.SHA1, 42)));
+
+                // TripleDes3KeyPkcs12 only works with SHA1
+                AssertExtensions.Throws<CryptographicException>(() =>
+                    export(slhDsa, "password", new PbeParameters(PbeEncryptionAlgorithm.TripleDes3KeyPkcs12, HashAlgorithmName.SHA512, 42)));
+            });
+
+            SlhDsaTestHelpers.AssertEncryptedExportPkcs8PrivateKey(export =>
+            {
+                // Bytes not allowed in TripleDes3KeyPkcs12
+                AssertExtensions.Throws<CryptographicException>(() =>
+                    export(slhDsa, "password", new PbeParameters(PbeEncryptionAlgorithm.TripleDes3KeyPkcs12, HashAlgorithmName.SHA1, 42)));
+            }, SlhDsaTestHelpers.EncryptionPasswordType.Byte);
+        }
 
         private const int PaddingSize = 10;
 
         [Theory]
-        [MemberData(nameof(ApiWithDestinationSpanTestData))]
-        public static void ExportSlhDsaPublicKey_CallsCore(SlhDsaAlgorithm algorithm, bool destinationLargerThanRequired)
+        [MemberData(nameof(SlhDsaTestData.AlgorithmsData), MemberType = typeof(SlhDsaTestData))]
+        public static void ExportSlhDsaPublicKey_CallsCore(SlhDsaAlgorithm algorithm)
         {
             using SlhDsaMockImplementation slhDsa = SlhDsaMockImplementation.Create(algorithm);
 
             int publicKeySize = algorithm.PublicKeySizeInBytes;
             byte[] publicKey = CreatePaddedFilledArray(publicKeySize, 42);
 
             // Extra bytes in destination buffer should not be touched
-            int extraBytes = destinationLargerThanRequired ? PaddingSize / 2 : 0;
-            Memory<byte> destination = publicKey.AsMemory(PaddingSize, publicKeySize + extraBytes);
+            Memory<byte> destination = publicKey.AsMemory(PaddingSize, publicKeySize);
```

Comment:
Don't forget to undo the test changes related to the signing changes that are being undone.